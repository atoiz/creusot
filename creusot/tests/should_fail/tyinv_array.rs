extern crate creusot_contracts;
use creusot_contracts::{invariant::Invariant, *};

pub struct Sum10(i32, i32);

impl Invariant for Sum10 {
    #[predicate]
    #[open]
    fn invariant(self) -> bool {
        pearlite! { self.0@ + self.1@ == 10 }
    }
}

pub fn take_array<const N: usize>(a: [Sum10; N]) -> [Sum10; N] {
    a
}

pub fn test_array() {
    let x = Sum10(3, 7);
    let a = [x];
    take_array(a);
}
