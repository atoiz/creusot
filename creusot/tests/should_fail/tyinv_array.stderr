error[creusot]: type invariants for arrays with generic length are not supported!

error: aborting due to previous error

